File: app/src/main/java/org/jellyfin/mobile/downloads/DownloadMethod.java
package org.jellyfin.mobile.downloads;

import static org.jellyfin.mobile.downloads.DownloadMethod.MOBILE_AND_ROAMING;
import static org.jellyfin.mobile.downloads.DownloadMethod.MOBILE_DATA;
import static org.jellyfin.mobile.downloads.DownloadMethod.WIFI_ONLY;

import androidx.annotation.IntDef;

@IntDef({WIFI_ONLY, MOBILE_DATA, MOBILE_AND_ROAMING})
public @interface DownloadMethod {
    int WIFI_ONLY = 0;
    int MOBILE_DATA = 1;
    int MOBILE_AND_ROAMING = 2;
}

File: app/src/main/java/org/jellyfin/mobile/downloads/DownloadMethod.kt
package org.jellyfin.mobile.downloads

enum class DownloadMethod(val intValue: Int) {
    WIFI_ONLY(0),
    MOBILE_DATA(1),
    MOBILE_AND_ROAMING(2),
    ;

    companion object {
        val DEFAULT = WIFI_ONLY

        fun fromInt(value: Int): DownloadMethod? = entries.find { it.intValue == value }
    }
}
